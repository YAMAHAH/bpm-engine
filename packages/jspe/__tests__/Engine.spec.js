import JSPE from 'jspe';

const jspe = new JSPE();

describe('Engine', () => {
  const consoleError = console.error;

  beforeEach(() => {
    console.error = () => {};
  });

  afterEach(() => {
    console.error = consoleError;
  });

  it('Can not instantiate a new processInstance without a workflowDefinition', async () => {
    await expect(jspe.createProcessInstance()).rejects.toMatchSnapshot();
  });

  it('Can not instantiate a new processInstance without a "valid" workflowDefinition', async () => {
    await expect(jspe.createProcessInstance({ workflowDefinition: 'INVALID_XML' })).rejects.toMatchSnapshot();
  });
});
