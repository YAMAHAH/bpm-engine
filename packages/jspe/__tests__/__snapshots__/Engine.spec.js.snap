// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Engine Can not instantiate a new processInstance without a "valid" workflowDefinition 1`] = `
[Error: unparsable content detected
	line: 0
	column: 0
	nested error: missing start tag]
`;

exports[`Engine Can not instantiate a new processInstance without a workflowDefinition 1`] = `[Error: Parameter "workflowDefinition" is required]`;
